### PR TITLE
PUB-569 - Search Error Screens

### PR DESCRIPTION
**Pull request by commitBlob**
### JIRA link ###
https://tools.hmcts.net/jira/browse/PUB-569

### Change description ###
Implement error screens for search option page

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```
invalid input screen
no-input-screen

### CODE REVIEW

**Review by johnoneill46 — APPROVED** [review]

File: src/main/views/search.njk [hunk 1]
```diff
@@ -1,48 +1,79 @@
...
+            <h1 class="govuk-label-wrapper">
+                <label class="govuk-label govuk-label--l" for="search-input">
+                    What court or tribunal are you interested in?
+                </label>
+            </h1>
+            <div id="search-input-hint" class="govuk-hint">
+                For example Blackburn Family Court or Chester
+            </div>
+            {% if noResultsError or invalidInputError %}
+                <span class="govuk-error-message">Enter a court name, town or city</span>
```
Comment (ChrisS1512) [inline src/main/views/search.njk:45]: Just on the error message - the user flow in the story says it should say 'Please enter a court or tribunal', however in the screenshots the above is correct (Enter a court name, town or city).

Is the above the correct one to be displayed?

Comment (commitBlob) [reply src/main/views/search.njk:45]: I was led by the screenshots from the ticket attachments

Comment (commitBlob) [reply src/main/views/search.njk:45]: I was led by screenshots from the ticket attachments

File: src/main/views/search.njk [hunk 2]
```diff
@@ -1,48 +1,79 @@
...
-  </script>
+    <script src="/assets/js/accessible-autocomplete.min.js"></script>
+    <script>
+      const element = document.querySelector('#search-input-container')
+      const source = {{ optionList | dump |safe }}
+        accessibleAutocomplete({
+          element: element,
+          id: 'search-input',
+          source: source,
+          autoselect: true
```
Comment (kurt-uk) [inline src/main/views/search.njk:76]: do we still need this if were not restricting people to searching by court name?

File: src/main/controllers/SearchController.ts [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
   public get(req: Request, res: Response): void {
-    const autocompleteList = courtList.getCourtsList();
-    res.render('search', {autocompleteList: autocompleteList});
+    res.render('search', { autocompleteList: autocompleteList, invalidInputError: false, noResultsError: false });
   }
 
   public post(req: Request, res: Response): void {
     const searchInput = req.body['input-autocomplete'];
-    res.redirect(`search-results?search-input=${searchInput}`);
+    if (searchInput && searchInput.length >= 3) {
```
Comment (kurt-uk) [inline src/main/controllers/SearchController.ts:17]: entering nothing doesnt seem to be rendering the error

**Review by kurt-uk — CHANGES_REQUESTED** [review]
dont forget to update the views test to make sure you can see the errors

File: src/main/views/search.njk (hunk 1 above)
Comment (commitBlob) [reply src/main/views/search.njk:45]: Junior confirmed it should be Enter a court name, town or city, updated the ticket accordingly

File: src/main/views/search.njk (hunk 2 above)
Comment (commitBlob) [reply src/main/views/search.njk]: removed autoselect

**Review by kurt-uk — APPROVED** [review]

**Pull request merged**